### PR TITLE
Drop db operations from hook commands

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: modules/private/branch.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"encoding/json"
+	"fmt"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// GetProtectedBranchBy get protected branch information
+func GetProtectedBranchBy(repoID int64, branchName string) (*models.ProtectedBranch, error) {
```
Comment (bkcsoft) [inline modules/private/branch.go:18]: This should be moved to `go-gitea/sdk` later on... I'm not gonna ask you to do it _now_ , but we're gonna have to move it at some point.

**Review by bkcsoft — COMMENTED** [review]
Just documenting this

File: modules/private/branch.go (hunk 1 above)
Comment (lunny) [reply modules/private/branch.go:18]: Why should we move it to `go-gitea/sdk` later? It is an internal SDK.

File: routers/private/push_update.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// license that can be found in the LICENSE file.
+
+package private
+
+import (
+	"encoding/json"
+	"strings"
+
+	macaron "gopkg.in/macaron.v1"
+
```
Comment (appleboy) [inline routers/private/push_update.go]: local package -> public package.

Comment (lunny) [reply routers/private/push_update.go]: done.

**Comment by lunny** [comment]
related #1023

File: cmd/hook.go [hunk 3]
```diff
@@ -64,6 +62,14 @@ var (
 	}
 )
 
+func hookSetup(logPath string) error {
+	setting.NewContext()
+	log.NewGitLogger(filepath.Join(setting.LogRootPath, logPath))
+	models.LoadConfigs()
+
+	return nil
```
Comment (appleboy) [inline cmd/hook.go]: No error return. Maybe changed to 

```go
func hookSetup(logPath string) {
```

Comment (lunny) [reply cmd/hook.go]: done.

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by andreynering** [comment]
@lunny Why do you think this is a good idea? Just trying to get the intention...

Doing GET and POST from hooks seems unstable to me. Will they work when HTTPS is being used?

**Comment by bkcsoft** [comment]
> Will they work when HTTPS is being used?

@andreynering They will use `localhost:{port}` AFAIK. Otherwise it's extremly fragile yes 😄

File: modules/private/branch.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	// Ask for running deliver hook and test pull request tasks.
+	reqURL := setting.LocalURL + fmt.Sprintf("api/internal/branch/%d/%s", repoID, branchName)
```
Comment (bkcsoft) [inline modules/private/branch.go:20]: @andreynering As you can see here it uses `LocalURL` which is fine 🙂

File: modules/private/branch.go (hunk 1 above)
Comment (bkcsoft) [reply modules/private/branch.go:18]: Should still be in the SDK. That way other implementations of `gitea serv` could re-use this :)

**Comment by bkcsoft** [comment]
@andreynering As far as "is this a good idea?", this is actually how `gitlab-shell` works as well, which seems to work just fine 😉

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

**Comment by b-k-c-soft** [comment]
LGTM

File: cmd/hook.go [hunk 5]
```diff
@@ -7,20 +7,18 @@ package cmd
 import (
```
Comment (b-k-c-soft) [inline cmd/hook.go]: idiots

Comment (b-k-c-soft) [inline cmd/hook.go]: it should be include

**Comment by lunny** [comment]
@andreynering It will let us avoid open db on `gitea serv`, `gitea hook pre-receive`, `gitea hook update` and `gitea hook post-receive` four times.

File: modules/private/branch.go (hunk 1 above)
Comment (sapk) [reply modules/private/branch.go:18]: I don't think that the sdk contain any internal route but we could maybe add it in an other time.

Comment (sapk) [reply modules/private/branch.go:18]: In fact it add /api/v1 to url request  (https://github.com/go-gitea/go-sdk/blob/master/gitea/gitea.go#L42) so no internal route.

**Comment by sapk** [comment]
LGTM

**Pull request merged**